### ISSUE TITLE
nx affected:lint -> Error: spawnSync /bin/sh ENOBUFS

### ISSUE THREAD

**Issue by dean-g** [nrwl/nx#5789]
why my nx affected query is failing after the angular 12 upgrades? 

please see the errors below,

```
nx affected:lint
Lint projects affected by changes
Run command using --base=[SHA1] (affected by the committed, uncommitted and
untracked changes):
  --base  Base of the current branch (usually master)                   [string]
or using --base=[SHA1] --head=[SHA2] (affected by the committed changes):
  --base  Base of the current branch (usually master)                   [string]
  --head  Latest commit of the current branch (usually HEAD)            [string]
or using:
  --files        Change the way Nx is calculating the affected command by
                 providing directly changed files, list of files delimited by
                 commas                                                  [array]
  --uncommitted  Uncommitted changes                                   [boolean]
  --untracked    Untracked changes                                     [boolean]
Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --parallel       Parallelize the command (default: false)            [boolean]
  --maxParallel    Max number of parallel processes. This flag is ignored if the
                   parallel option is set to `false`. (default: 3)      [number]
  --all            All projects                                        [boolean]
  --exclude        Exclude certain projects from being processed
                                                           [array] [default: []]
  --runner         This is the name of the tasks runner configured in nx.json
                                                                        [string]
  --skip-nx-cache  Rerun the tasks even when the results are available in the
                   cache                              [boolean] [default: false]
  --configuration  This is the configuration to use when performing tasks on
                   projects                                             [string]
  --only-failed    Isolate projects which previously failed
                                                      [boolean] [default: false]
  --verbose        Print additional error stack trace on failure
<ref *1> Error: spawnSync /bin/sh ENOBUFS
    at Object.spawnSync (internal/child_process.js:1067:20)
    at spawnSync (child_process.js:602:24)
    at Object.execSync (child_process.js:647:15)
    at parseGitOutput (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:64:28)
    at getUntrackedFiles (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:43:12)
    at Object.parseFiles (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:33:20)
    at projectsToRun (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/affected.js:97:108)
    at Object.<anonymous> (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/affected.js:27:26)
    at Generator.next (<anonymous>)
    at fulfilled (/builds/platform/ui/client/node_modules/@nrwl/workspace/node_modules/tslib/tslib.js:114:62) {
  errno: -105,
  code: 'ENOBUFS',
  syscall: 'spawnSync /bin/sh',
  path: '/bin/sh',
  spawnargs: [ '-c', 'git ls-files --others --exclude-standard' ],
  error: [Circular *1],
  status: null,
  signal: 'SIGTERM',
  output: [
    null,
    <Buffer 2e 6e 70 6d 2f 73 65 6e 74 72 79 2d 63 6c 69 2f 33 66 63 35 30 36 2d 73 65 6e 74 72 79 2d 63 6c 69 2d 4c 69 6e 75 78 2d 78 38 36 2d 36 34 0a 63 61 63 ... 10244046 more bytes>,
    <Buffer >
  ],
  pid: 551,
  stdout: <Buffer 2e 6e 70 6d 2f 73 65 6e 74 72 79 2d 63 6c 69 2f 33 66 63 35 30 36 2d 73 65 6e 74 72 79 2d 63 6c 69 2d 4c 69 6e 75 78 2d 78 38 36 2d 36 34 0a 63 61 63 ... 10244046 more bytes>,
  stderr: <Buffer >
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Cleaning up file based variables
00:02
ERROR: Job failed: exit code 1x affected:lint
Lint projects affected by changes
Run command using --base=[SHA1] (affected by the committed, uncommitted and
untracked changes):
  --base  Base of the current branch (usually master)                   [string]
or using --base=[SHA1] --head=[SHA2] (affected by the committed changes):
  --base  Base of the current branch (usually master)                   [string]
  --head  Latest commit of the current branch (usually HEAD)            [string]
or using:
  --files        Change the way Nx is calculating the affected command by
                 providing directly changed files, list of files delimited by
                 commas                                                  [array]
  --uncommitted  Uncommitted changes                                   [boolean]
  --untracked    Untracked changes                                     [boolean]
Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --parallel       Parallelize the command (default: false)            [boolean]
  --maxParallel    Max number of parallel processes. This flag is ignored if the
                   parallel option is set to `false`. (default: 3)      [number]
  --all            All projects                                        [boolean]
  --exclude        Exclude certain projects from being processed
                                                           [array] [default: []]
  --runner         This is the name of the tasks runner configured in nx.json
                                                                        [string]
  --skip-nx-cache  Rerun the tasks even when the results are available in the
                   cache                              [boolean] [default: false]
  --configuration  This is the configuration to use when performing tasks on
                   projects                                             [string]
  --only-failed    Isolate projects which previously failed
                                                      [boolean] [default: false]
  --verbose        Print additional error stack trace on failure
<ref *1> Error: spawnSync /bin/sh ENOBUFS
    at Object.spawnSync (internal/child_process.js:1067:20)
    at spawnSync (child_process.js:602:24)
    at Object.execSync (child_process.js:647:15)
    at parseGitOutput (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:64:28)
    at getUntrackedFiles (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:43:12)
    at Object.parseFiles (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/shared.js:33:20)
    at projectsToRun (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/affected.js:97:108)
    at Object.<anonymous> (/builds/platform/ui/client/node_modules/@nrwl/workspace/src/command-line/affected.js:27:26)
    at Generator.next (<anonymous>)
    at fulfilled (/builds/platform/ui/client/node_modules/@nrwl/workspace/node_modules/tslib/tslib.js:114:62) {
  errno: -105,
  code: 'ENOBUFS',
  syscall: 'spawnSync /bin/sh',
  path: '/bin/sh',
  spawnargs: [ '-c', 'git ls-files --others --exclude-standard' ],
  error: [Circular *1],
  status: null,
  signal: 'SIGTERM',
  output: [
    null,
    <Buffer 2e 6e 70 6d 2f 73 65 6e 74 72 79 2d 63 6c 69 2f 33 66 63 35 30 36 2d 73 65 6e 74 72 79 2d 63 6c 69 2d 4c 69 6e 75 78 2d 78 38 36 2d 36 34 0a 63 61 63 ... 10244046 more bytes>,
    <Buffer >
  ],
  pid: 551,
  stdout: <Buffer 2e 6e 70 6d 2f 73 65 6e 74 72 79 2d 63 6c 69 2f 33 66 63 35 30 36 2d 73 65 6e 74 72 79 2d 63 6c 69 2d 4c 69 6e 75 78 2d 78 38 36 2d 36 34 0a 63 61 63 ... 10244046 more bytes>,
  stderr: <Buffer >
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Cleaning up file based variables
00:02
ERROR: Job failed: exit code 1
```

**Comment by dean-g** [comment]
@isaacplmann @brandonroberts @juristr Can you please advise me on this issue ?

**Comment by brandonroberts** [comment]
@dean-g can you provide more information about your environment? Are other affected commands working?

**Comment by dean-g** [comment]
I got the same error for nx affected:test as well. this is happening only on my gitlab environment. but the same working good in my local machine!.

**Comment by dean-g** [comment]
the same worked well before the angular 12 upgrades. very wired what's causing this from angular 12 upgrades.

**Comment by dean-g** [comment]
And this issue is happening when we have more affected changes. if there is no much affected changed, my build got to succeed. not sure this is something to deal with the cache. please advise.

**Comment by vsavkin** [comment]
@dean-g could you provide a repro?

**Comment by dean-g** [comment]
@vsavkin sorry. I can't share that info. it was private

**Comment by dean-g** [comment]
the build is working if I kept on retrying (2 to 3 times). I am not sure what could be the rootcause. please advise

**Comment by dean-g** [comment]
I am also getting Out of memory issues often from ng build commend. 

`error Command failed with exit code 137.` (Out of memory)

very wired!. all these happening only after angular 12 migration!

**Comment by dean-g** [comment]
I am also getting the following error occasionally, 

```
An unhandled exception occurred: Call retries were exceeded
See "/tmp/ng-opaQQn/angular-errors.log" for further details.
error Command failed with exit code 127.
error Command failed with exit code 127.

```

**Comment by kevinolivar** [comment]
I updated nx yesterday and started getting the same issue running the following command in gitlab ci/cd,

yarn affected:lint --base=origin/master

package.json
    "@nrwl/cli": "12.3.5",
    "@nrwl/cypress": "12.3.5",
    "@nrwl/eslint-plugin-nx": "12.3.5",
    "@nrwl/express": "^12.3.5",
    "@nrwl/jest": "12.3.5",
    "@nrwl/linter": "12.3.5",
    "@nrwl/next": "12.3.5",
    "@nrwl/node": "12.3.5",
    "@nrwl/react": "12.3.5",
    "@nrwl/storybook": "12.3.5",
    "@nrwl/tao": "12.3.5",
    "@nrwl/web": "12.3.5",
    "@nrwl/workspace": "12.3.5",

**Comment by Coly010** [comment]
@kevinolivar Hi!

@dean-g mentioned he couldn't provide a repo that repro's this issue.

@kevinolivar Would you be able to provide us with one that we could use to investigate?

**Comment by kevinolivar** [comment]
@Coly010 Hi! Unfortunately, it is also on a private repo. Sorry. I thought maybe yarn cache was the culprit. I moved out of the workspace and that did not work as well.
I am thinking maybe some circular dependencies somewhere or JSON cycle?

Environment

    "@nrwl/cli": "12.3.5",
    "@nrwl/cypress": "12.3.5",
    "@nrwl/eslint-plugin-nx": "12.3.5",
    "@nrwl/express": "^12.3.5",
    "@nrwl/jest": "12.3.5",
    "@nrwl/linter": "12.3.5",
    "@nrwl/next": "12.3.5",
    "@nrwl/node": "12.3.5",
    "@nrwl/react": "12.3.5",
    "@nrwl/storybook": "12.3.5",
    "@nrwl/tao": "12.3.5",
    "@nrwl/web": "12.3.5",
    "@nrwl/workspace": "12.3.5",

**Comment by kevinolivar** [comment]
Hi there,

just to give an update in case someone else comes across this issue as well.

Coincidently, I was able to fix this error by updating the cache and install in my gitlab-ci.yml.

Just to verify, I then logged the output of `git ls-files --others --exclude-standard` and there was no more node_modules in the listing. Reducing the output to below 10 Megabytes.

Below the setup,

```
.install: &install "[ ! -d node_modules ] && time CYPRESS_CACHE_FOLDER='./.cache/Cypress' yarn install --frozen-lockfile --prefer-offline --cache-folder .yarn || echo yarn.lock unchanged. Using cache."

cache: &use_yarn_cache
  key:
    files:
      - yarn.lock
  untracked: true
  policy: pull
  paths:
    - .yarn
    - node_modules
    - .cache/Cypress
````

I am not 100% sure what happened before but I assumed it was because the way my gitlab-ci.yml was configured. Eg. node_modules was not cached properly...

Anyway, I am still looking forward to getting an answer for my previous comment.

Cheers
